esmf_regrid/_esmf_sdo.py

```diff
@@ -138,21 +149,37 @@ def __init__(
         self.circular = circular
         self.areas = areas
         super().__init__(
-            shape=(len(lats), len(lons)),
+            shape=shape,
             index_offset=1,
             field_kwargs={"staggerloc": ESMF.StaggerLoc.CENTER},
         )
 
     def _as_esmf_info(self):
         shape = np.array(self._shape)
 
-        if self.circular:
-            adjustedlonbounds = self.lonbounds[:-1]
-        else:
-            adjustedlonbounds = self.lonbounds
-
-        centerlons, centerlats = np.meshgrid(self.lons, self.lats)
-        cornerlons, cornerlats = np.meshgrid(adjustedlonbounds, self.latbounds)
+        londims = len(self.lons.shape)
+        assert len(self.lons.shape) == londims
+        latdims = len(self.lats.shape)
+        assert len(self.lats.shape) == latdims
+        assert londims == latdims
+        assert londims in (1, 2)
+
+        if londims == 1:
+            if self.circular:
+                adjustedlonbounds = self.lonbounds[:-1]
+            else:
+                adjustedlonbounds = self.lonbounds
+            centerlons, centerlats = np.meshgrid(self.lons, self.lats)
+            cornerlons, cornerlats = np.meshgrid(adjustedlonbounds, self.latbounds)
+        elif londims == 2:
+            if self.circular:
+                slice = np.s_[:, :-1]
```

Comment:
coverage says that this line is not being hit by the tests. so I assume that means we don't have a test case for circular curvilinear data. should we?

Reply:
So it turns out that only `DimCoord`s have the `circular` property, so this is not going to be hit via the iris scheme. With that said, it is theoretically possible to call this by setting up a regridder directly. In that sense the code here is complete with respect to the core of esmf-regrid and the testing is complete with respect to the iris schemes but there is now a hole in the testing with respect to the core of esmf-regrid. I suppose it's probably worth adding a test for this case for completeness sake.